core/src/main/java/io/grpc/ClientInterceptors.java

```diff
@@ -204,4 +212,190 @@ public final void start(Listener<RespT> responseListener, Metadata headers) {
       }
     }
   }
+
+
+  public static class RetryingInterceptor implements ClientInterceptor {
+    // TODO(lukaszx0) have default logger for this one to give more visibility into when/what gets
+    // retried?
+    // private final Logger logger = Logger.getLogger(RetryingInterceptor.class.getName());
+    private final RetryPolicy retryPolicy;
+
+    public RetryingInterceptor(RetryPolicy retryPolicy) {
+      this.retryPolicy = retryPolicy;
+    }
+
+    public RetryingInterceptor() {
+      this.retryPolicy = new IdempotentRetryPolicy();
+    }
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+            MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+      return new RetryingCall<ReqT, RespT>(method, callOptions, next, retryPolicy);
```

Comment:
Should just return `next.newCall(method, callOptions)` if this is not a unary RPC call. (We could fix it up to support streaming... but that gets a *lot* more complicated and wouldn't work on bidi-streaming methods that want to implement real bidirectional messaging (vs. streaming upload followed by streaming download).